### PR TITLE
Refactors

### PR DESCRIPTION
**Pull request by princemaple**
close https://github.com/firezone/product/issues/348

The one usage in this PR is now justified. And I think I changed another one in one of my other PRs.

The rest are all good to keep.

### CODE REVIEW

File: apps/fz_http/lib/fz_http/shared_validators.ex [hunk 1]
```diff
@@ -35,77 +32,49 @@ defmodule FzHttp.SharedValidators do
...
-
-        []
-      catch
-        ip ->
-          [{field, "is invalid: #{String.trim(ip)} is not a valid FQDN or IPv4 / IPv6 address"}]
-      end
-    end)
-  end
-
-  def validate_ip(changeset, field) when is_atom(field) do
```
Comment (princemaple) [inline apps/fz_http/lib/fz_http/shared_validators.ex:53]: This is not used

**Comment by princemaple** [comment]
blocked by https://github.com/firezone/firezone/pull/648/commits/b2437f79f337769d5c2353c91aaddc88b70ca8dc

File: apps/fz_http/lib/fz_http/users/user.ex [hunk 2]
```diff
@@ -162,17 +162,13 @@ defmodule FzHttp.Users.User do
     |> cast(attrs, [:email, :last_signed_in_method, :last_signed_in_at])
   end
 
-  def authenticate_user(user, password_candidate) do
-    Argon2.check_pass(user, password_candidate)
-  end
```
Comment (princemaple) [inline apps/fz_http/lib/fz_http/users/user.ex:167]: not used outside the module

**Review by jamilbk — APPROVED** [review]
LGTM

**Pull request merged**